updater/updater.go

```diff
@@ -133,6 +135,9 @@ func Run(config *config.UpdaterConfig, datastore database.Datastore, st *utils.S
 					nextUpdate = lockExpiration
 				}
 			}
+		} else {
+			DbFinishUpdate = false
```

Comment:
@AndreeaDora You may make use of the available `firstUpdate` variable. While not entirely safe (the updater doesn't keep its distributed state in a specialized service such as etcd), it is better already.